src/metabase/driver/sql_jdbc/connection.clj

```diff
@@ -274,6 +290,12 @@
                     ;; don't log the actual spec lest we accidentally expose credentials
                     {:input (class db-or-id-or-spec)}))))
 
+(defn db->pooled-connection-spec
+  "Returns a JDBC connection spec that includes a cp30 `ComboPooledDataSource`.
+  Similar to [[db+connection-type->pooled-connection-spec]] but uses the `::r` connection type by default."
+  [db-or-id-or-spec]
+  (db+connection-type->pooled-connection-spec db-or-id-or-spec ::r))
```

Comment:
Instead of having this **and** `db+connection-type->pooled-connection-spec` as a separate function, wouldn't it be easier just to change this to have an additional arity?

e.g.

```clj
(defn db->pooled-connection-spec
 ([db-or-id-or-spec]
  (db->pooled-connection-spec db-or-id-or-spec :readonly))
 ([db-or-id-or-spec readonly-or-readwrite)
  ...)
```